client/src/components/TaskFilterSelector/TaskFilterSelector.js

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import styles from "./TaskFilterSelector.module.css";
+import { PropTypes } from "prop-types";
+
+const FilterButton = ({ filter, filterSelected, onFilterChanged }) => {
+  return (
+    <button
+      className={
+        filter === filterSelected ? styles.selectedFilter : styles.filterButton
+      }
+      onClick={() => onFilterChanged(filter)}
+    >
+      {filter}
+    </button>
+  );
+};
+
+FilterButton.propTypes = {
+  filter: PropTypes.string.isRequired,
+  filterSelected: PropTypes.string.isRequired,
+  onFilterChanged: PropTypes.func.isRequired,
+};
+
+export const TaskFilterSelector = ({ filterSelected, onFilterChanged }) => {
+  const filters = ["Day", "Week", "Month", "All"];
+  const filterButtons = filters.map((filter, ix) => (
+    <FilterButton
+      key={ix}
+      filter={filter}
```

Comment:
I would also rename this prop to be `filterName`

Reply:
Renamed.